### PR TITLE
Add more skip check options for Proxy objects

### PR DESCRIPTION
**Pull request by zherczeg**
Reorganize the flags to follow the list in ES2020 section 9.5

### CODE REVIEW

**Comment by zherczeg** [comment]
I am thinking whether it is a good idea to create a flag for each handler or creating groups would be better. Or create only one flag.

**Comment by zherczeg** [comment]
I have been thinking about this, and I am not sure there are use cases where you want to disable checks for `get` but not for `delete`. The point is, the user declartes that the Proxy is trusted, and no need sanity checks. Or the `target` is more like the prototype of the Proxy, so the checks are invalid.

**Review by rerobika — APPROVED** [review]

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -94,8 +94,8 @@ These option bits allow specializing Proxies with non-standard behaviour.
 These flags are recommended only for those trusted Proxies, whose handlers
 produce correct results.
 
-- JERRY_PROXY_SKIP_GET_CHECKS - skip [[Get]] result checks
-- JERRY_PROXY_SKIP_GET_OWN_PROPERTY_CHECKS - skip [[GetOwnProperty]] result checks
+- JERRY_PROXY_SKIP_RESULT_VALIDATION - skip result validation for [[IsExtensible]], [[PreventExtensions]], [[GetOwnProperty]],
+                                       [[DefineOwnProperty]], [[HasProperty]], [[Get]], [[Set]], [[Delete]] and [[OwnPropertyKeys]]
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md:98]: What about the Get/SetPrototype other validations? Shouldn't we add those also?

Comment (zherczeg) [reply docs/02.API-REFERENCE.md]: I have added the prototype handlers, but that should be enough.

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**